src/App.tsx

```diff
@@ -49,34 +50,30 @@ export const App: React.FC = () => {
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector value={author} onChange={setAuthor} />
+                <UserSelector value={author} onChange={handleSetAuthor} />
               </div>
 
               <div className="block" data-cy="MainContent">
                 {!author && <p data-cy="NoSelectedUser">No user selected</p>}
-
-                {author && !loaded && <Loader />}
-
-                {author && loaded && hasError && (
+                {author && !isPostsLoading && <Loader />}
+                {author && isPostsLoading && postsError && (
                   <div
                     className="notification is-danger"
                     data-cy="PostsLoadingError"
                   >
                     Something went wrong!
                   </div>
                 )}
-
-                {author && loaded && !hasError && posts.length === 0 && (
+                {author && isPostsLoading && !postsError && !posts.length && (
                   <div className="notification is-warning" data-cy="NoPostsYet">
                     No posts yet
                   </div>
                 )}
-
-                {author && loaded && !hasError && posts.length > 0 && (
+                {author && isPostsLoading && !postsError && !!posts.length && (
```

Comment:
It is a good practice to create variables for such long conditions for better readability